hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/TestFairScheduler.java

```diff
@@ -3369,7 +3369,48 @@ public void testBasicDRFWithQueues() throws Exception {
     scheduler.handle(updateEvent);
     Assert.assertEquals(1, app2.getLiveContainers().size());
   }
-  
+
+  @Test
+  public void testDRFWithClusterResourceChanges() throws Exception {
```

Comment:
Using a "real" scheduler with mock nodes seems excessive for this. Can we just mock the scheduler and context? Also, this might be a good test for TestDRF than TestFairScheduler.